src/vs/workbench/parts/files/common/editors/textFileEditorModel.ts

```diff
@@ -441,66 +441,81 @@ export class TextFileEditorModel extends BaseTextEditorModel implements ITextFil
 		// A save participant can still change the model now and since we are so close to saving
 		// we do not want to trigger another auto save or similar, so we block this
 		// In addition we update our version right after in case it changed because of a model change
+		let saveParticipantPromise = TPromise.as(versionId);
+
 		if (TextFileEditorModel.saveParticipant) {
-			this.blockModelContentChange = true;
-			try {
-				TextFileEditorModel.saveParticipant.participate(this, { isAutoSaved });
-			} finally {
+			saveParticipantPromise = TPromise.as(undefined).then(() => {
+				this.blockModelContentChange = true;
+				return TextFileEditorModel.saveParticipant.participate(this, { isAutoSaved });
+			}).then(() => {
 				this.blockModelContentChange = false;
-			}
-			versionId = this.versionId;
+				return this.versionId;
+			}, err => {
+				this.blockModelContentChange = false;
+				return TPromise.wrapError(err);
```

Comment:
@jrieken this is a semantical change from how it worked before where a save participant error would never make the save operation fail. maybe we should ignore the possible errors from save participants (especially since they can now come from extensions) and always allow the save to take place.